shopinvader_search_engine/models/product_template.py

```diff
@@ -20,6 +20,9 @@ def _get_parent_categories(self, categ_ids):
         return categories
 
     @api.depends_context("index_id")
-    @api.depends("categ_id", "categ_id.parent_id")
+    @api.depends(lambda self: self._compute_shopinvader_category_depends())
```

Comment:
To extend the `depends` you just need to override the method with a new `api.depends` decorator with only your new dependent. The list of depends is computed by collecting all the depends declared across the class hierarchy. see (https://github.com/odoo/odoo/blob/16.0/odoo/fields.py#L568)

Reply:
Yes but I find this way more explicit and flexible. It's also a well known pattern in the Odoo ecosystem. That's why I proposed it.

Reply:
It depends... Both are valid. The `override approach has the advantage of allowing you to easily find all `depends` for a given field using a regular expression...

Reply:
The final question is 'What guidelines do we want to promote to ensure that the same patterns are used across all shopinvader modules, and to ensure consistency?' @qgroulard @AnizR @marielejeune @sebastienbeau

Reply:
I agree with @lmignon, both approaches are valid but I personally prefer the 'override' approach with new @api.depends

Reply:
using @api.depends() is enough, I do not see case that are not solve with this solution.